lib/std/builtin/assembly.zig

```diff
@@ -0,0 +1,1026 @@
+        h1: bool = false,
+        h2: bool = false,
+        h3: bool = false,
+        h4: bool = false,
+        h5: bool = false,
+        h6: bool = false,
+        h7: bool = false,
+        h8: bool = false,
+        h9: bool = false,
+        h10: bool = false,
+        h11: bool = false,
+        h12: bool = false,
+        h13: bool = false,
+        h14: bool = false,
+        h15: bool = false,
+        h16: bool = false,
+        h17: bool = false,
+        h18: bool = false,
+        h19: bool = false,
+        h20: bool = false,
+        h21: bool = false,
+        h22: bool = false,
+        h23: bool = false,
+        h24: bool = false,
+        h25: bool = false,
+        h26: bool = false,
+        h27: bool = false,
+        h28: bool = false,
+        h29: bool = false,
+        h30: bool = false,
+        h31: bool = false,
+
+        b0: bool = false,
+        b1: bool = false,
+        b2: bool = false,
+        b3: bool = false,
+        b4: bool = false,
+        b5: bool = false,
+        b6: bool = false,
+        b7: bool = false,
+        b8: bool = false,
+        b9: bool = false,
+        b10: bool = false,
+        b11: bool = false,
+        b12: bool = false,
+        b13: bool = false,
+        b14: bool = false,
+        b15: bool = false,
+        b16: bool = false,
+        b17: bool = false,
+        b18: bool = false,
+        b19: bool = false,
+        b20: bool = false,
+        b21: bool = false,
+        b22: bool = false,
+        b23: bool = false,
+        b24: bool = false,
+        b25: bool = false,
+        b26: bool = false,
+        b27: bool = false,
+        b28: bool = false,
+        b29: bool = false,
+        b30: bool = false,
+        b31: bool = false,
+
+        za0: bool = false,
+        za1: bool = false,
+        za2: bool = false,
+        za3: bool = false,
+        za4: bool = false,
+        za5: bool = false,
+        za6: bool = false,
+        za7: bool = false,
+        za8: bool = false,
+        za9: bool = false,
+        za10: bool = false,
+        za11: bool = false,
+        za12: bool = false,
+        za13: bool = false,
+        za14: bool = false,
+        za15: bool = false,
+
+        zt0: bool = false,
+    },
+    .arm, .armeb => packed struct {
+        /// Whether the inline assembly code may perform stores to memory
+        /// addresses other than those derived from input pointer provenance.
+        memory: bool = false,
+
+        apsr: bool = false,
+        cpsr: bool = false,
+        spsr: bool = false,
+        r0: bool = false,
+        r1: bool = false,
+        r2: bool = false,
+        r3: bool = false,
+        r4: bool = false,
+        r5: bool = false,
+        r6: bool = false,
+        r7: bool = false,
+        r8: bool = false,
+        r9: bool = false,
+        r10: bool = false,
+        r11: bool = false,
+        r12: bool = false,
+        r13: bool = false,
+        r14: bool = false,
+        r15: bool = false,
```

Comment:
`r15` is the program counter. I would argue the same thing here as for `rip` on x86.